zio-http/src/main/scala/zhttp/api/API.scala

```diff
@@ -0,0 +1,87 @@
+package zhttp.api
+
+import zhttp.http.{Headers => _, Path => _}
+import zio.schema.Schema
+
+/**
+ *   - Input and Output as Schemas.
+ *   - Support wide range of Codecs including Json
+ *     - Dynamically decide response format based upon Request Header
+ */
+final case class API[Params, Input, Output](
+  method: HttpMethod,
+  requestParser: RequestParser[Params], // Path / QueryParams / Headers
+  doc: Doc,
+  inputSchema: Schema[Input],           // Generate any sort of codec, generate OpenAPI docs.
+  outputSchema: Schema[Output],
```

Comment:
```suggestion
  decoder: BodyDecoder[Input],           // Generate any sort of codec, generate OpenAPI docs.
  encoder: BodyEncoder[Output],
```

Should we avoid `Schema` here? To support Scala 3 and reduce onboarding cost for new devs?

Reply:
We need something Schema in order to support OpenAPI generation, as well as the ability to encode/decode to different formats based on headers. But perhaps for a first release, until Schema supports Scala 3.